### PR TITLE
feat: add json pointer and level information in context

### PR DESCRIPTION
**Pull request by jy95**
https://github.com/jy95/docusaurus-json-schema-plugin/discussions/223 and #120 

@gnidan Like I explained in https://github.com/jy95/docusaurus-json-schema-plugin/discussions/223#discussioncomment-7983729 , Having a second pair of eyes could help ;) You have the preview (with its playground) on Vercel to see how it works with your contrived example and others as well

image

```json
{
	"$schema": "http://json-schema.org/draft-07/schema",
	"allOf": [
		{
			"oneOf": [
				{
					"type": "object",
					"properties": {
						"foo": {
							"const": "a"
						}
					}
				}
			]
		}
	]
}
```

I think we can release this part and later a solution for `CreateNodes` missing id in html case

### CODE REVIEW

File: src/theme/JSONSchemaViewer/JSONSchemaElements/SchemaConditional/if-else-then.tsx [hunk 1]
```diff
@@ -47,11 +53,38 @@ export default function IfElseThen(props: Props): JSX.Element {
...
+        )
       case "schema_then":
-        return <CreateNodes schema={schema.then!} />
+        return (
+          <SchemaHierarchyContextProvider
+            value={{
+              level: currentLevel + 1,
+              jsonPointer: `${currentJsonPointer}/then`,
+            }}
+          >
```
Comment (gnidan) [inline src/theme/JSONSchemaViewer/JSONSchemaElements/SchemaConditional/if-else-then.tsx:73]: In my implementation, I avoided the need to `useSchemaHierarchyContext` everywhere by building one of these:

```tsx
export type InnerSchemaHierarchyContextProviderProps = {
  innerJsonPointer: `/${string}`;
  children: JSX.Element;
}

export function InnerSchemaHierarchyContextProvider({
  innerJsonPointer,
  children
}: InnerSchemaHierarchyContextProvider): JSX.Element {
  const { jsonPointer, level } = useSchemaHierarchyContext()

  return (
    <SchemaHierarchyContextProvider value={{
      jsonPointer: `${jsonPointer}${innerJsonPointer}`,
      level: level + 1
    }}>
      {children}
    </SchemaHierarchyContextProvider>
  )
}
```

Thought you might find this nice :)

**Comment by gnidan** [comment]
Besides the above suggestion, I'm going to test this tonight by swizzling components in my schema docs repo, to try to get my desired behavior. Your implementation generally looks good to me (about the same as I had, except I got stuck on the allOfs/etc.)

File: src/theme/JSONSchemaViewer/JSONSchemaElements/SchemaConditional/if-else-then.tsx (hunk 1 above)
Comment (jy95) [reply src/theme/JSONSchemaViewer/JSONSchemaElements/SchemaConditional/if-else-then.tsx]: Thanks for the idea. I called it `SchemaHierarchyComponent` and found a missing pointer for [`contentSchema`](https://opis.io/json-schema/2.x/string.html#contentschema) (quite unused in the real world but we never know ...)

**Comment by jy95** [comment]
@gnidan 1.10 version will be out so you be free to swizzle it ;)

**Pull request merged**